WordPress/gutenberg
# Fix inserting button block when pressing enter in a block with bound `text` attribute

## What?
Discussed in [this other pull request](https://github.com/WordPress/gutenberg/pull/58958#issuecomment-1956667124).

If a button block text is bound to a custom field, it disables its editing. However, when you try to add a new button by pressing enter when the bound button is selected, nothing happens. It should add a new button the same way it does with not connected ones. To reproduce it you can:

1. Add a button connected to a custom field in the editor.

```html

<div class="wp-block-buttons">
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Text</a></div>
</div>

```

2. Select the button and press Enter.
3. Check that nothing happens.


This pull request tries to solve that.

## Why?
The current UX is not the expected one.

## How?
There are two issues that need to be solved:

1. The button block rich text is not focusable when it has bindings.

I'm solving this by always setting the tabindex to 0 when the rich text is bound: [link](https://github.com/WordPress/gutenberg/commit/6554533d590ea2ceb0f02ec470b91b04909563ba).

2. The button block should insert a new button when it is selected.

I'm solving that part by manually inserting a new block when the rich text is not editable and the user press Enter: [link](https://github.com/WordPress/gutenberg/commit/1bad234cc9a403cd5f7f90eb653627dfce123ec4). Not 100% sure this is the right approach but it seems to be working for the current use cases (paragraph, heading, and button).

## Testing Instructions
1. Add a button connected to a custom field in the editor.

```html

<div class="wp-block-buttons">
<div class="wp-block-button"><a class="wp-block-button__link wp-element-button">Text</a></div>
</div>



```

2. Select the button and press Enter.
3. Check that a new empty button has been added.

Repeat the process with a paragraph and a heading and check they work as expected as well.
